xapian-core/stemmer/KrovetzStemmer.cc

```diff
@@ -0,0 +1,6846 @@
+/*==========================================================================
+ * Copyright (c) 2005 University of Massachusetts.  All Rights Reserved.
+ *
+ * Use of the Lemur Toolkit for Language Modeling and Information Retrieval
+ * is subject to the terms of the software license set forth in the LICENSE
+ * file included with this software, and also available at
+ * http://www.lemurproject.org/license.html
```

Comment:
Unfortunately that licence has a clause restricting naming of derived products which makes it incompatible with the GPL:

>   4. Products derived from this software may not be called "Lemur" or "Indri" nor may "Lemur" or "Indri" appear in their names without prior written permission of The Lemur Project. To obtain permission, contact license@lemurproject.org.

(Based on https://www.gnu.org/licenses/license-list.html#Phorum having the equivalent clause and that making that licence incompatible with the GPL.)

It's very annoying that something so minor makes it incompatible.  There are similar issues with the PHP licence and the GPL, so you can't legally distribute builds of Xapian's PHP bindings and everyone has to build it for themselves instead.

I think that sadly means we can't merge this, though it's still useful for getting your more familiar with the code.